### ISSUE TITLE
nomaps: report pointer value types in maps

### ISSUE THREAD

**Issue by mattsu2020** [kubernetes-sigs/kube-api-linter#250]
## Problem statement

While working on kubernetes-sigs/controller-tools#1416, I added a controller-gen warning for map types whose value type is a pointer, for example:

```go
type ExampleSpec struct {
    Values map[string]*string `json:"values,omitempty"`
}
```

The controller-tools PR implemented this as a non-fatal code generation diagnostic, but review feedback suggested that this kind of API-convention finding should likely live in kube-api-linter instead of controller-tools.

kube-api-linter already has a `nomaps` linter that checks usage of map types and enforces Kubernetes API design guidance around maps. It seems like the closest fit would be to extend `nomaps` so it can also report map value types that are pointers.

## Why this is useful

Pointer values in maps are supported by Go and controller-gen, but they are not a great Kubernetes API shape. They can introduce ambiguous API semantics around omitted map entries vs present keys with nil values, and they make the schema/API harder to reason about.

Catching this in kube-api-linter would let API authors get feedback during linting, without adding API-convention warnings to controller-tools/code generation.

## Proposed behavior

Add a `nomaps` check for map value types that are pointers, such as:

```go
map[string]*string
map[string]*SomeStruct
map[string]*metav1.Time
```

Example diagnostic:

```text
nomaps: map value type should not be a pointer; prefer a non-pointer value type or remodel the API shape
```

Potential configuration option:

```yaml
lintersConfig:
  nomaps:
    disallowPointerValues: true
```

or, if this should follow existing policy-style configuration:

```yaml
lintersConfig:
  nomaps:
    pointerValuesPolicy: Enforce | Warn | Ignore
```

The exact configuration shape is open to maintainer preference.

## Acceptance criteria

* `nomaps` reports a violation for `map[string]*T` value types when the check is enabled.
* `map[string]string` remains allowed under the existing `AllowStringToStringMaps` behavior.
* `map[string]T` non-pointer value types continue to follow the existing `nomaps.policy` behavior.
* Tests cover both pointer value and non-pointer value cases.
* Documentation for `nomaps` configuration is updated.

## Related context

* controller-tools PR: https://github.com/kubernetes-sigs/controller-tools/pull/1416
* The PR originally attempted to add a controller-gen warning for pointer map values, but review feedback pointed toward kube-api-linter as the better home for this finding.

**Comment by JoelSpeed** [comment]
> They can introduce ambiguous API semantics around omitted map entries vs present keys with nil values, and they make the schema/API harder to reason about.

Could you provide an example (via go playground?) that demonstrates how pointer maps don't round trip correctly, if it's a round tripping issue (which I think is what I'm picking up on) then I'm supportive of adding it to a linter here
